scripts/airgapped/retag-images-to-cache.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+import logging
+
+import docker
+
+from utils import (delete_file_if_exists, get_images_list_from_file,
+                   get_or_pull_image)
+
+cli = docker.client.from_env()
+
+log = logging.getLogger(__name__)
+
+RETAGGED_IMAGES_FILE = "retagged-images.txt"
+SHA_TOKEN = "@sha256"
+REGISTRIES = ["docker.io/library", "docker.io", "gcr.io", "quay.io",
+              "registry.k8s.io", "k8s.gcr.io", "public.ecr.aws", "ghcr.io",
+              "nvcr.io"]
+
+
+def retag_image_with_sha(image):
+    """Retag the image by using the sha value."""
+    log.info("Retagging image digest: %s", image)
+    repo_digest = image.attrs["RepoDigests"][0]
+    [repository_name, sha_value] = repo_digest.split("@sha256:")
+
+    tagged_image = "%s:%s" % (repository_name, sha_value)
+    log.info("Retagging to: %s", tagged_image)
+    image.tag(tagged_image)
+
+    log.info("Tagged image successfully: %s", tagged_image)
+    return cli.images.get(tagged_image)
+
+
+def get_retagged_image_name(image_nm: str, new_registry: str) -> str:
+    """Given an image name replace the repo and use sha as tag."""
+    if SHA_TOKEN in image_nm:
+        log.info("Provided image has sha. Using it's value as tag.")
+        image_nm = image_nm.replace(SHA_TOKEN, "")
+
+    if len(image_nm.split("/")) == 1:
+        # docker.io/library image, i.e. ubuntu:22.04
+        return "%s/%s" % (new_registry, image_nm)
+
+    if len(image_nm.split("/")) == 2:
+        # classic docker.io image, i.e. argoproj/workflow-controller
+        return "%s/%s" % (new_registry, image_nm)
```

Comment:
Good observation, but the code avoids this bug 😄 

If there was just one `/` (so after split we have 2 strings) then we will just append the new registry at the beginning. So in case of such docker.io images they'll end up from:
`argoproj/workflow-controller:tag` -> `new-registry/argoproj/workflow-controller:tag`